src/software/replay/replay_reader.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include <experimental/filesystem>
+
+#include "software/proto/tbots_replay.pb.h"
+#include "software/proto/sensor_msg.pb.h"
+
+class ReplayReader {
+   public:
+    explicit ReplayReader(const std::string& _replay_dir);
+    SensorMsg getNextFrame();
```

Comment:
could combine `getNextFrame()` and `hasNextFrame()` into `std::optional<SensorMsg> getNextFrame()`.

Reply:
it was like that before last night, but i thought being able to do
```
while (reader.hasNextFrame()) {
auto frame = reader.getNextFrame();
// do stuff with the frame
}
```
would be cleaner

Reply:
Technically you could do
```
while(auto frame = reader.getNextFrame()) {
	// do stuff with the frame
}
```

I'm mostly suggesting this since it's consistent with what we do most of the time (we tend to use more optionals with our getter functions and don't have `hasNext...` functions since it's slightly more "functional-programming" style). I don't have a super strong opinion here.

Reply:
totally forgot we could do that, changed